### ISSUE TITLE
thanos query-frontend web.external-prefix

### ISSUE THREAD

**Issue by ebini** [thanos-io/thanos#4351]
### Describe the solution you'd like

i would like to have
thanos query frontend also to have the option "web.external-prefix"

to make it accessible behind a reverse proxy

at the moment i get (v0.21.1)
`thanos: error: unknown long flag '--web.external-prefix'`

Thanks in advance

**Comment by yeya24** [comment]
Valid request. Help wanted.

**Comment by matej-g** [comment]
@yeya24 happy to take this one

**Comment by yeya24** [comment]
> @yeya24 happy to take this one

Nice. Please go ahead!

**Comment by matej-g** [comment]
I think this is still a valid request, although unfortunately I didn't have the time to address it. 

From my understanding, this is a bit more complicated to do for Query Frontend, since it's more of a rounder tripper and does not expose it's own UI.

**Comment by ataahua** [comment]
Add me to the list of requestors. Very useful potential feature.

**Comment by jphx** [comment]
This is still a desired enhancement.

**Comment by alialperak** [comment]
Due to nature of **query-frontend** actually you don't need a prefix configuration here. You need to set **--web.external-prefix** for query component only to make it work behind a reverse proxy.

**Comment by jphx** [comment]
Yes, and I'd like to deploy query-frontend to make it available behind a proxy also.  But I can't at the moment because it doesn't support a prefix.  See https://github.com/thanos-io/thanos/issues/4351#issuecomment-944502946.

**Comment by alialperak** [comment]
@jphx my answer is same for your use case also. You need to set `--web.external-prefix` for `query` component only to make `query-frontend` work behind a reverse proxy. You can thank me later.

**Comment by jphx** [comment]
@alialperak I do that currently, @alialperak, but it doesn't solve the problem.  I use a router that routes requests based on the path prefix.  The query component supports using a prefix, so I can correctly route requests to the query component, but the frontend component doesn't, so I can't route requests to the frontend.  I could choose to route the query component's prefix to the frontend instead of the query component, making the query component unreachable, but since we have a genuine need to make the query component available as well, that doesn't solve the problem for us.
